src/construction.cpp

```diff
@@ -2221,10 +2242,12 @@ void load_construction( const JsonObject &jo )
     }
 
     jo.read( "pre_note", con.pre_note );
-    con.pre_terrain = jo.get_string( "pre_terrain", "" );
-    if( con.pre_terrain.size() > 1
-        && con.pre_terrain[0] == 'f'
-        && con.pre_terrain[1] == '_' ) {
+    con.pre_terrain = jo.get_as_string_set( "pre_terrain" );
+    const std::string &first_pre_terrain = *con.pre_terrain.begin();
```

Comment:
This needs to test that pre_terrain is not empty, otherwise it dereferences the end iterator which is invalid.

Reply:
Good catch. https://github.com/CleverRaven/Cataclysm-DDA/pull/78372